### ISSUE TITLE
[0.9.8] flipx issues

### ISSUE THREAD

**Issue by Kaillera** [Regalis11/Barotrauma#2634]
- [x] I have searched the issue tracker to check if the issue has already been reported.

**Description**
Using flipx in singleplayer makes the sub disappears, and water pressure killl all crew members.
Using flipx in sub editor makes the game crash.

[crashreport (3).log](https://github.com/Regalis11/Barotrauma/files/4482915/crashreport.3.log)

**Version**
0.9.8

**Additional Informations**
Used The Aeche (III) as test

**Comment by Regalis11** [comment]
Fixed the crashing in https://github.com/Regalis11/Barotrauma-development/commit/0ec279c

Could not reproduce the sub disappearing in single player.

**Comment by Kaillera** [comment]
[Uploading The Aeche (III).zip…]()

Try flipx on this sub.

**Comment by Regalis11** [comment]
Thanks, managed to figure it out now! I only tested this on vanilla subs, but it seems it only happened when there's several subs linked to the main one.

Should be fixed in https://github.com/Regalis11/Barotrauma-development/commit/c89ca18

**Comment by Rokvach** [comment]
Doing a flipx while the Aeche is docked to a outpost in singleplayer caused a crash, no issues once the sub is undocked @Regalis11 

In Body.cs
![image.png](https://images.zenhubusercontent.com/5c924117534a9b0a86cdddce/db91e530-e01b-46b3-9da9-930f0f8528f3)

```
Barotrauma Client crash report (generated on 4.5.2020 15:07:04)


Barotrauma seems to have crashed. Sorry for the inconvenience! 


0EF58219431435DD725FF1F0C02A87C4


Game version 0.9.9.1 (ReleaseWindows, branch dev, revision 195c3b940)
Graphics mode: 2560x1440 (BorderlessWindowed)
VSync OFF
Language: English
Selected content packages: Vanilla 0.9, The Aeche (III)
Level seed: Hyperenor LineaUndermoist
Loaded submarine: The Aeche (III) (F8DB0521CC0CB4ECB152555B1342D2A1)
Selected screen: Barotrauma.GameScreen
SteamManager initialized


System info:
    Operating system: Microsoft Windows NT 10.0.18362.0 64 bit
    GPU name: NVIDIA GeForce GTX 980 Ti
    Display mode: {Width:2560 Height:1440 Format:Color AspectRatio:1,7777778}
    GPU status: Normal


Exception: Object reference not set to an instance of an object. (System.NullReferenceException)
Target site: Void Remove(FarseerPhysics.Dynamics.Fixture)
Stack trace: 
   at FarseerPhysics.Dynamics.Body.Remove(Fixture fixture) in C:\Users\RoniK\Documents\Barotrauma-development\Libraries\Farseer Physics Engine 3.5\Dynamics\Body.cs:line 651
   at Barotrauma.Items.Components.DockingPort.Undock() in C:\Users\RoniK\Documents\Barotrauma-development\Barotrauma\BarotraumaShared\SharedSource\Items\Components\DockingPort.cs:line 830
   at Barotrauma.Submarine.SetPosition(Vector2 position, List`1 checkd) in C:\Users\RoniK\Documents\Barotrauma-development\Barotrauma\BarotraumaShared\SharedSource\Map\Submarine.cs:line 911
   at Barotrauma.Submarine.FlipX(List`1 parents) in C:\Users\RoniK\Documents\Barotrauma-development\Barotrauma\BarotraumaShared\SharedSource\Map\Submarine.cs:line 787
   at Barotrauma.DebugConsole.<>c.<InitProjectSpecific>b__22_87(String[] args) in C:\Users\RoniK\Documents\Barotrauma-development\Barotrauma\BarotraumaClient\ClientSource\DebugConsole.cs:line 2203
   at Barotrauma.DebugConsole.Command.Execute(String[] args) in C:\Users\RoniK\Documents\Barotrauma-development\Barotrauma\BarotraumaShared\SharedSource\DebugConsole.cs:line 77
   at Barotrauma.DebugConsole.ExecuteCommand(String command)
   at Barotrauma.DebugConsole.Update(Single deltaTime) in C:\Users\RoniK\Documents\Barotrauma-development\Barotrauma\BarotraumaClient\ClientSource\DebugConsole.cs:line 178
   at Barotrauma.GameMain.Update(GameTime gameTime) in C:\Users\RoniK\Documents\Barotrauma-development\Barotrauma\BarotraumaClient\ClientSource\GameMain.cs:line 911
   at Microsoft.Xna.Framework.Game.DoUpdate(GameTime gameTime) in C:\Users\RoniK\Documents\Barotrauma-development\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 656
   at Microsoft.Xna.Framework.Game.Tick() in C:\Users\RoniK\Documents\Barotrauma-development\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 500
   at Microsoft.Xna.Framework.SdlGamePlatform.RunLoop() in C:\Users\RoniK\Documents\Barotrauma-development\Libraries\MonoGame.Framework\Src\MonoGame.Framework\SDL\SDLGamePlatform.cs:line 92
   at Microsoft.Xna.Framework.Game.Run(GameRunBehavior runBehavior) in C:\Users\RoniK\Documents\Barotrauma-development\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 397
   at Microsoft.Xna.Framework.Game.Run() in C:\Users\RoniK\Documents\Barotrauma-development\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 367
   at Barotrauma.Program.Main(String[] args) in C:\Users\RoniK\Documents\Barotrauma-development\Barotrauma\BarotraumaClient\ClientSource\Program.cs:line 58


Last debug messages:
[4.5.2020 15:07:03] flipx
[4.5.2020 15:07:02] Steam achievements have been disabled during this play session.
[4.5.2020 15:07:02] Enabled cheat commands.
[4.5.2020 15:07:02] enablecheats
[4.5.2020 15:06:48] Could not compress a texture because the dimensions are a multiple of 4 (path: C:/Users/RoniK/Documents/Barotrauma-development/Barotrauma/bin/ReleaseWindows/netcoreapp3.1/Content/Items/Electricity/ShuttleBattery.png, size: 897x307)
[4.5.2020 15:06:31] Content package load order: Vanilla 0.9  |  The Aeche (III)
[4.5.2020 15:06:29] OpenAL Soft on Mikrofoni (Blackstorm Scout ) 44
[4.5.2020 15:06:21] Content package load order: Vanilla 0.9
[4.5.2020 15:06:20] OpenAL Soft on Mikrofoni (Blackstorm Scout ) 44
[4.5.2020 15:05:26] Loaded languages: English, German, French, Russian, Polish, Brazilian Portuguese, Castilian Spanish, Latinamerican Spanish, Simplified Chinese, Traditional Chinese, Japanese, Turkish, Korean
[4.5.2020 15:05:26] Content package load order: Vanilla 0.9
[4.5.2020 15:05:26] Content package load order: Vanilla 0.9
[4.5.2020 15:05:24] Attempting to open ALC device "OpenAL Soft"
[4.5.2020 15:05:23] Loaded languages: English, German, French, Russian, Polish, Brazilian Portuguese, Castilian Spanish, Latinamerican Spanish, Simplified Chinese, Traditional Chinese, Japanese, Turkish, Korean
[4.5.2020 15:05:22] Loaded languages: English, German, French, Russian, Polish, Brazilian Portuguese, Castilian Spanish, Latinamerican Spanish, Simplified Chinese, Traditional Chinese, Japanese, Turkish, Korean
```

**Comment by Regalis11** [comment]
Fixed in 80b13ae

**Comment by Rokvach** [comment]
No issues, closing.
